flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/parser/JaninoCompiler.java

```diff
@@ -298,4 +312,33 @@ private static Java.Rvalue generateNoOperandTimestampFunctionOperation(String op
                 StringUtils.convertToCamelCase(operationName),
                 timestampFunctionParam.toArray(new Java.Rvalue[0]));
     }
+
+    private static String getTypeConvertMethod(SqlDataTypeSpec sqlDataTypeSpec) {
+        switch (sqlDataTypeSpec.getTypeName().getSimple().toUpperCase()) {
+            case "BOOLEAN":
+                return "convertBoolean";
+            case "TINYINT":
+                return "convertByte";
+            case "SMALLINT":
+                return "convertShort";
+            case "INTEGER":
+                return "convertInteger";
+            case "BIGINT":
+                return "convertLong";
+            case "FLOAT":
+                return "convertFloat";
+            case "DOUBLE":
+                return "convertDouble";
+            case "DECIMAL":
+                return "convertBigDecimal";
+            case "CHAR":
+            case "VARCHAR":
+            case "STRING":
+            case "`STRING`":
```

Comment:
Seems only the STRING type is specially treated to handle the quoted version. Will other types be quoted too? If so, maybe we can trim typename first before `switch` cases.